### ISSUE TITLE
corelib: int.modPow can produce wrong results for divisor greater than half-word

### ISSUE THREAD

**Issue by rakudrama** [dart-lang/sdk#37469]
The inner loop of `int.modPow` contains the statements `r = (r * b) % m;` and `b = (b * b) % m;`.
The problem is truncation in the product. 0 ≤ **b** < **m** and likewise **r**, so the intermediate product requires twice as many bits as **m**. This means that the wrong result may be computed for 64-bit integers on the VM when **m** has > 31 bits. Compiled to JavaScript the wrong results can appear if **m** has more than 26 bits.

dart2js/ddc currently fail corelib_2/int_modulo_arith_test/modPow since some test values for **m** have more than 26 bits. (The VM passes because the test values have 31 or fewer bits.)

The implementation could be changed to make use of a special product-modulo operation, but it might be simpler to use the BigInt implementation when the divisor is larger than the 'half-word', or perhaps always.

The ddc/dart2js implementation should probably error on inputs outside the 'safe integer' range - such inputs have already been subject to rounding and the exponentiation will propagate the rounding errors and produce a meaningless result.

If we have the opportunity we should probably deprecate and remove int.modPow. The main use case is RSA with large values that require BigInts.

**Comment by rakudrama** [comment]
Example.  `(M-1)**E mod M` is either `1` (E even) or `M-1` (E odd).
```
main() {
  demo(30000000001, 2, 30000000002);
  demo(30000000001, 3, 30000000002);
  demo(30000000001, 1000000, 30000000002);
}

demo(int a, int e, int m) {
  print('$a.modPow($e, $m)  =  ${a.modPow(e, m)}'
      '  vs  ${BigInt.from(a).modPow(BigInt.from(e), BigInt.from(m))}');
}
```
VM:
```
30000000001.modPow(2, 30000000002)  =  28491334789  vs  1
30000000001.modPow(3, 30000000002)  =  7439306033  vs  30000000001
30000000001.modPow(1000000, 30000000002)  =  779511127  vs  1
```
dart2js:
```
30000000001.modPow(2, 30000000002)  =  43008  vs  1
30000000001.modPow(3, 30000000002)  =  29999956994  vs  30000000001
30000000001.modPow(1000000, 30000000002)  =  16853595274  vs  1
```

**Comment by rakudrama** [comment]
Fixed via 41d3971e833e3d70c18cebee033ee80f347cebbe
